xcom2-launcher/xcom2-launcher/Classes/Mod/ModInfo.cs

```diff
@@ -46,59 +46,59 @@ protected void LoadFile(string filepath)
                     Contract.Assume(line != null);
 
                     // Expected format is "key=value"
-                    if (key == null || line.Contains("="))
+                    if (currentKey == null || line.Contains("="))
                     {
-                        var data = line.Split(new[] { '=' }, 2);
-                        var temp = data[0].Trim().ToLower();
+                        var keyValPair = line.Split(new[] { '=' }, 2);
+                        var newKey = keyValPair[0].Trim().ToLower();
 
-                        if (key == null || keys.Contains(temp))
+                        if (currentKey == null || validKeys.Contains(newKey))
                         {
                             // probably right
-                            key = temp;
-                            var val = data[1];
+                            currentKey = newKey;
+                            var currentValue = keyValPair[1];
 
                             // A multi-line value should be indicated by a trailing '\' (source?) but most of the time isn't !?
                             while (line.Last() == '\\')
                             {
                                 line = reader.ReadLine();
                                 Contract.Assume(line != null);
-                                val += "\r\n" + line;
+                                currentValue += "\r\n" + line;
                             }
 
-                            if (values.ContainsKey(key))
+                            if (keyValPairs.ContainsKey(currentKey))
                             {
                                 // This only happens if a tag appears multiple times.
-                                values[key] = val;
+                                keyValPairs[currentKey] = currentValue;
                             }
                             else
                             {
-                                values.Add(key, val);
+                                keyValPairs.Add(currentKey, currentValue);
```

Comment:
Why not just `keyValPairs[currentKey] = currentValue;` in all cases?

Reply:
That would throw KeyNotFoundException. Or maybe I am missing your point.

Reply:
![image](https://user-images.githubusercontent.com/2865341/66461334-7a986800-ea81-11e9-9778-36d79b1b54e5.png)

`KeyNotFoundException` is thrown when you attempt to read a non-existent key

Reply:
Interesting, I never realized that it only throws when reading. In that case we can simplify that.